modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Pagination.tsx

```diff
@@ -0,0 +1,187 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import classnames from 'classnames';
+import {fetch, openToast} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import {TFDSView} from '../FDSViews';
+import RequiredMark from '../RequiredMark';
+
+const FDS_PAGE_SIZES = '4, 8, 20, 40, 60';
+const FDS_DEFAULT_PAGE_SIZE = '20';
+
+interface IPaginationProps {
+	fdsView: TFDSView;
+	fdsViewsAPIURL: string;
+	namespace: string;
+}
+
+function Pagination({fdsView, fdsViewsAPIURL, namespace}: IPaginationProps) {
+	const [pageSizes, setPageSizes] = useState(FDS_PAGE_SIZES);
+	const [defaultPageSize, setDefaultPageSize] = useState(
+		FDS_DEFAULT_PAGE_SIZE
+	);
+	const [defaultValueError, setDefaultValueError] = useState(false);
+
+	useEffect(() => {
+		const paginationArray = pageSizes.split(', ');
+
+		if (!paginationArray.includes(defaultPageSize)) {
+			setDefaultValueError(true);
+		}
+		else {
+			setDefaultValueError(false);
+		}
+	}, [defaultPageSize, pageSizes]);
+
+	const handleSaveClick = async () => {
+		const body = {
+			defaultPageSize,
+			label: fdsView.label,
+			pageSizes,
+		};
+
+		const response = await fetch(
+			`${fdsViewsAPIURL}/by-external-reference-code/${fdsView.externalReferenceCode}`,
+			{
+				body: JSON.stringify(body),
+				headers: {
+					'Accept': 'application/json',
+					'Content-Type': 'application/json',
+				},
+				method: 'PATCH',
+			}
+		);
+
+		const responseJSON = await response.json();
+
+		if (responseJSON?.id) {
+			openToast({
+				message: Liferay.Language.get(
+					'your-request-completed-successfully'
+				),
+				type: 'success',
+			});
+		}
+		else {
+			openToast({
+				message: Liferay.Language.get(
+					'your-request-failed-to-complete'
+				),
+				type: 'danger',
+			});
+		}
+	};
+
+	return (
+		<ClayLayout.Sheet className="mt-3" size="lg">
+			<ClayLayout.SheetHeader>
+				<h2 className="sheet-title">
+					{Liferay.Language.get('pagination')}
+				</h2>
+
+				<p>
+					Use values separated by commas to generate different
+					pagination items in the Dataset view.{' '}
+				</p>
+			</ClayLayout.SheetHeader>
+
+			<ClayLayout.SheetSection>
+				<ClayForm.Group>
+					<label htmlFor={`${namespace}fdsViewPageSizesTextarea`}>
+						{Liferay.Language.get('page-sizes')}
+
+						<RequiredMark />
+					</label>
+
+					<ClayInput
+						component="textarea"
+						id={`${namespace}fdsViewPaginationItemsTextarea`}
+						onChange={(event) => setPageSizes(event.target.value)}
+						required
```

Comment:
Is this enough? :thinking: 

Required behavior is:
- `has-error` class on container
- error message `this-field-is-required`

So far, we implemented this manually, see [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L323-L354).